restapi/src/main/java/io/stargate/web/docsapi/service/query/provider/ConditionProviderService.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.service.query.provider;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.web.docsapi.service.query.condition.BaseCondition;
+import io.stargate.web.docsapi.service.query.predicate.impl.*;
+
+import java.util.*;
+
+public class ConditionProviderService {
+
+    private final Map<String, ConditionProvider> providerMap;
+
+    public ConditionProviderService() {
+        this.providerMap = initProviderMap();
+    }
+
+    public Collection<BaseCondition> getConditions(JsonNode conditionsNode, boolean numericBooleans) {
+        List<BaseCondition> results = new ArrayList<>();
+        Iterator<Map.Entry<String, JsonNode>> fields = conditionsNode.fields();
+        fields.forEachRemaining(field -> {
+            String filterOp = field.getKey();
+            ConditionProvider jsonNodeConditionProvider = providerMap.get(filterOp);
+            if (null != jsonNodeConditionProvider) {
+                Optional<? extends BaseCondition> condition = jsonNodeConditionProvider.createCondition(field.getValue(), numericBooleans);
+                condition.ifPresent(results::add);
+            }
+
+            // TODO what in case fo the non found, f.e $my, throw exception?
```

Comment:
Meaning if an invalid filter is sent over the wire? This probably should throw and cause a 400 upstream

Reply:
I would agree. However, I just realized that there is no "mechanism" of throwing exceptions in the service that can later on be transformed to some kind of the status code + provide correct `Error` back. Usually I deal with such situations, by having something like this:

```
class BusinessRuntimeException extends RuntimeException {

  private String message;
  private ErrorCode errorCode;

  public Response getResponse() {
     ResponseStatus status = errorCode.getStatusProvider().get();
     String defaultMessage = errorCode.getStatusProvider().getDefaultMesssage();
     Error entity = new Error(Optional.ofNullable(message).orElse(defaultMessage), status.getCode());
     return Response.status(status)
        .entity(error)
       .build():
  }
}

enum ErrorCode {
   
   Supplier<ResponseStatus> statusSupplier;
   String defaultMessage;

   DOCSAPI_FILTER_WRONG_ARGUMENT(() -> ResponseStatus.BAD_REQUEST),
   DOCSAPI_FILTER_OPERATION_DOES_NOT_EXIST(() -> ResponseStatus.BAD_REQUEST),
   ....
   ....
}
```

Let's discuss this, would be a nice addition to the complete Rest API. This is also very nice for testing, as you can always confirm from the service that thrown exception has proper `ErrorCode`.

Reply:
I think this could be really nice to have, and extends well to anything that works over HTTP (maybe even graphql, though typically the HTTP status code is always 200 - but I could imagine it using the Exception's default message to make a response)

The only thing I could see becoming unwieldy with this setup is that the enum `ErrorCode` could get pretty long, but for the most part I feel like that can be fixed by keeping each enum value in some kind of order.

Reply:
In separate ticket.